replay/models/ddpg.py

```diff
@@ -504,13 +622,19 @@ def _batch_pass(self, batch: dict) -> Dict[str, Any]:
         next_state = self.model.state_repr(next_user, next_memory)
         next_action = self.target_model(next_user, next_memory)
         target_value = self.target_value_net(next_state, next_action.detach())
-        expected_value = reward + (1.0 - done) * self.gamma * target_value
+        expected_value = (
+            reward + (1.0 - done) * self.gamma * target_value
+        )  # smth strange, check article
         expected_value = torch.clamp(
             expected_value, self.min_value, self.max_value
         )
 
         value = self.value_net(state, action)
-        value_loss = (value - expected_value.detach()).squeeze(1).pow(2).mean()
+        value_loss = (
+            (((value - expected_value.detach())).pow(2) * sample_weight)
```

Comment:
is this weighting related to BayesUCBDQN?

Reply:
Yes